pkg/kvstore/etcd.go

```diff
@@ -381,25 +381,24 @@ func (e *etcdClient) closeLockSession(leaseID client.LeaseID) {
 	e.RWMutex.RUnlock()
 }
 
-func (e *etcdClient) waitForInitLock(ctx context.Context) <-chan bool {
-	initLockSucceeded := make(chan bool)
+func (e *etcdClient) waitForInitLock(ctx context.Context) <-chan error {
+	initLockSucceeded := make(chan error)
 
 	go func() {
 		for {
 			select {
 			case <-e.client.Ctx().Done():
-				initLockSucceeded <- false
+				initLockSucceeded <- fmt.Errorf("client context ended")
```

Comment:
similar to what is done on line 395, why not `fmt.Errorf("client context ended: %w", e.client.Ctx().Err())`?